novawallet/Modules/ManualBackupKeyList/ManualBackupKeyListPresenter.swift

```diff
@@ -0,0 +1,200 @@
+    func didReceive(_ chainsChange: [DataProviderChange<ChainModel>]) {
+        chains = chainsChange.mergeToDict(chains)
+
+        let sortedChains = sorted(
+            chains,
+            for: metaAccount
+        )
+        let viewModel = createViewModel(from: sortedChains)
+
+        view?.update(with: viewModel)
+    }
+}
+
+// MARK: Private
+
+private extension ManualBackupKeyListPresenter {
+    func createViewModel(from sortedChains: SortedChains) -> ManualBackupKeyListViewLayout.Model {
+        let listHeaderText = R.string.localizable.chainAccountsListHeader(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        return .init(
+            listHeaderText: listHeaderText,
+            accountsSections: [
+                createDefaultChainsSection(for: sortedChains.defaultChains),
+                createCustomChainsSection(for: sortedChains.customChains)
+            ]
+        )
+    }
+
+    func createDefaultChainsSection(for chains: [ChainModel]) -> ManualBackupKeyListViewLayout.Sections {
+        let defaultChainsHeaderText = R.string.localizable.chainAccountsListDefaultHeader(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        let defaultChainsTitleText = R.string.localizable.chainAccountsListDefaultTitle(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        return .defaultKeys(
+            .init(
+                headerText: defaultChainsHeaderText.uppercased(),
+                accounts: [
+                    .init(
+                        title: defaultChainsTitleText,
+                        subtitle: formattedString(for: chains)
+                    )
+                ]
+            )
+        )
+    }
+
+    func createCustomChainsSection(for chains: [ChainModel]) -> ManualBackupKeyListViewLayout.Sections {
+        let customChainsViewModels = chains
+            .compactMap { [weak self] chain -> ManualBackupKeyListViewLayout.CustomAccount? in
+                guard let self else { return .none }
+
+                return ManualBackupKeyListViewLayout.CustomAccount(
+                    network: networkViewModelFactory.createViewModel(from: chain),
+                    chainId: chain.chainId
+                )
+            }
+
+        let customChainsHeaderText = R.string.localizable.chainAccountsListCustomHeader(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        return .customKeys(
+            .init(
+                headerText: customChainsHeaderText.uppercased(),
+                accounts: customChainsViewModels
+            )
+        )
+    }
+
+    func sorted(
+        _ chains: [ChainModel.Id: ChainModel],
+        for metaAccount: MetaAccountModel
+    ) -> SortedChains {
+        let defaultChainsIds = Set(metaAccount.chainAccounts.map(\.chainId))
```

Comment:
Why it is called defaultChainsIds if it holds custom chain ids?

Reply:
My bad, forgot to rename it after few rewrites 😅